python/cudf/cudf/core/frame.py

```diff
@@ -4755,7 +4755,7 @@ def _scan(self, op, axis=None, skipna=True, cast_to_int=False):
                 result_col = self._data[name].nans_to_nulls()
             else:
                 result_col = self._data[name].copy()
-                if result_col.has_nulls:
+                if result_col.has_nulls or result_col.has_nans:
```

Comment:
Can we be sure that `result_col` is of a numeric type and that the call to `has_nans` will always succeed?

Reply:
yup, any other numeric types other than `float` will get `nan_count` as 0 which will result in `has_nans` to be `False`.

Reply:
Had a discussion offline and decided to introduce `include_nan` to `has_nulls` (method- formerly a property).